src/Workspaces/Core/Portable/FindSymbols/Shared/AbstractSyntaxIndex_Persistence.cs

```diff
@@ -70,12 +94,17 @@ internal partial class AbstractSyntaxIndex<TIndex> : IObjectWritable
             // We also want the checksum to change any time our serialization format changes.  If
             // the format has changed, all previous versions should be invalidated.
             var project = document.Project;
-            var parseOptionsChecksum = project.State.GetParseOptionsChecksum();
```

Comment:
So no concern about the cache now being valid for other parse options changing? You mentioned the langver record parse difference of behavior; doc comment mode also changes parsing, any concerns there? Any things in indexes are based on span, right? Not node indicies (i.e. 2nd node in the file)?

If we think it this is fine it might be worth a callout here that the behavior is intentional; I could imagine potential changes (caching syntax diagnostics or something) that _might_ care.

Reply:
> So no concern about the cache now being valid for other parse options changing? You mentioned the langver record parse difference of behavior; doc comment mode also changes parsing, any concerns there?

I'll doc that this is intentional.  Here's why:

1. only 'record' parses differently because of lang ver.  But there's no realistic (e.g. non-pathological) scenario where you could have the file be correct under two different lang versions where both had different, but legal meaning.   So i'm fine just taking the meaning from one langversion and assuming the code has the same meaning in the other.
2. for doc comments we *always* at least parse in the IDE, so you always get the same parse tree.  


> Not node indicies (i.e. 2nd node in the file)?

Correct.

Reply:
doc'ed